include/ginkgo/core/base/mpi.hpp

```diff
@@ -413,7 +426,9 @@ class communicator {
      * @param color The color to split the original comm object
      * @param key  The key to split the comm object
      */
-    communicator(const MPI_Comm& comm, int color, int key)
+    communicator(const MPI_Comm& comm, int color, int key,
+                 std::shared_ptr<const Executor> exec)
```

Comment:
In this file, the `exec` now needs to be added to the documentations

Reply:
Overall, the communicator documentation needs to be overhauled a bit.